pkgs/tools/virtualization/xe-guest-utilities/default.nix

```
@@ -1,82 +1,39 @@
-  patches = [ ./ip-address.patch ];
+
+  vendorHash = "sha256-zhpDvo8iujE426/gxJY+Pqfv99vLNKHqyMQbbXIKodY=";
+
   postPatch = ''
-    tar xf "$NIX_BUILD_TOP/$name/xenstore-sources.tar.bz2"
+    substituteInPlace Makefile \
+      --replace 'RELEASE := $(shell git rev-list HEAD | wc -l)' ""
   '';
 
-  # Workaround build failure on -fno-common toolchains:
-  #   ld: utils.o:xenstore/utils.h:27:
-  #     multiple definition of `xprintf'; xenstored_core.o:xenstore/utils.h:27: first defined here
-  NIX_CFLAGS_COMPILE = "-fcommon";
-
   buildPhase = ''
-    export CC=gcc
-    export CFLAGS='-Wall -Wstrict-prototypes -Wno-unused-local-typedefs -Wno-sizeof-pointer-memaccess'
-    export PYTHON=python2
-    cd "$NIX_BUILD_TOP/$name/uclibc-sources"
-    for file in Config.mk tools/libxc/Makefile tools/misc/Makefile tools/misc/lomount/Makefile tools/xenstore/Makefile; do
-      substituteInPlace "$file" --replace -Werror ""
-    done
-    make -C tools/include
-    make -C tools/libxc
-    make -C tools/xenstore
+    make
   '';
 
   installPhase = ''
-    export CFLAGS+='-Wall -Wstrict-prototypes -Wno-unused-local-typedefs -Wno-sizeof-pointer-memaccess'
-    if [[ $CARCH == x86_64 ]]; then
-      export LIBLEAFDIR_x86_64=lib
-    fi
-    for f in include libxc xenstore; do
-      [[ ! -d $NIX_BUILD_TOP/$name/uclibc-sources/tools/$f ]] && continue
-      make -C "$NIX_BUILD_TOP/$name/uclibc-sources/tools/$f" DESTDIR="$out" BINDIR=/bin SBINDIR=/bin INCLUDEDIR=/include LIBDIR=/lib install
-    done
-    rm -r "$out"/var
-
-    cd "$NIX_BUILD_TOP/$name"
-    install -Dm755 xe-update-guest-attrs "$out/bin/xe-update-guest-attrs"
-    install -Dm755 xe-daemon "$out/bin/xe-daemon"
-    install -Dm644 xen-vcpu-hotplug.rules "$out/lib/udev/rules.d/10-xen-vcpu-hotplug.rules"
-    substituteInPlace "$out/bin/xe-daemon" --replace sbin bin
-    substituteInPlace "$out/bin/xe-daemon" --replace /usr/ "$out/"
-    substituteInPlace "$out/bin/xe-update-guest-attrs" --replace /usr/ "$out/"
-    substituteInPlace "$out/bin/xe-update-guest-attrs" --replace 'export PATH=' 'export PATH=$PATH:'
-    substituteInPlace "$out/lib/udev/rules.d/10-xen-vcpu-hotplug.rules" --replace /bin/sh '${bash}/bin/sh'
-
-    cat <<'EOS' >"$out/bin/xe-linux-distribution"
-    #!${bash}/bin/bash -eu
-    . /etc/os-release
-    if [[ $# -gt 0 ]]; then
-      mkdir -p "$(dirname "$1")"
-      exec 1>"$1"
-    fi
-    cat <<EOF
-    os_distro="$ID"
-    os_majorver="''${VERSION_ID%%.*}"
-    os_minorver="''${VERSION_ID#*.}"
-    os_uname="$(uname -r)"
-    os_name="$PRETTY_NAME"
-    EOF
-    EOS
-    chmod 0755 "$out/bin/xe-linux-distribution"
+    install -Dt $out/bin build/stage/usr/{,s}bin/*
   '';
 
-})
+  meta = {
+    description = "XenServer guest utilities";
+    homepage = "https://github.com/xenserver/xe-guest-utilities";
+    license = lib.licenses.bsd2;
+    maintainers = with lib.maintainers; [ ];
+    platforms = lib.platforms.linux;
+  };
+}
```

Since we are already rewriting the whole file, we might as well format it a little bit

> XenServer guest utilities for unix-like operating systems

might be able to change platforms to `unix` as well

```suggestion
  meta = with lib; {
    description = "XenServer guest utilities";
    homepage = "https://github.com/xenserver/xe-guest-utilities";
    license = licenses.bsd2;
    maintainers = with maintainers; [ ];
    platforms = platforms.linux;
  };
}
```

---

I try to avoid nesting `with`.

---

> might be able to change platforms to `unix` as well

ofborg build failed on Darwin, so I changed it back to linux.